pkg/configmap/configmap.go

```diff
@@ -0,0 +1,166 @@
+func NewConfigMapsService(client core.ConfigMapInterface, helper kubernetes.Kubernetes) ConfigMapsService {
+	return ConfigMapsService{
+		client: client,
+		helper: helper,
+	}
+}
+
+// ListNamesAndLabels return names and labels of Config Maps
+func (cms ConfigMapsService) ListNamesAndLabels() (resourceNames, labels []string, err error) {
+	configMaps, err := cms.List(metav1.ListOptions{})
+	var objectMetas []metav1.ObjectMeta
+	for _, cm := range configMaps {
+		objectMetas = append(objectMetas, cm.ObjectMeta)
+	}
+	configMapNames, labels := util.GetNamesAndLabels(objectMetas)
+	return configMapNames, labels, nil
+}
+
+// List returns a list of ConfigMaps from a namespace
+func (cms ConfigMapsService) List(listOptions metav1.ListOptions) ([]v1.ConfigMap, error) {
+	configMaps, err := cms.client.List(listOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	return configMaps.Items, nil
+}
+
+// GetUnused return unused Config Maps
+func (cms ConfigMapsService) GetUnused(namespace string, configMaps []v1.ConfigMap) (unusedConfigMaps []v1.ConfigMap, funcErr error) {
+	var usedConfigMaps []v1.ConfigMap
+	funk.ForEach(openshift.PredefinedResources, func(predefinedResource schema.GroupVersionResource) {
+		funk.ForEach(configMaps, func(resource v1.ConfigMap) {
+
+			resourceName := resource.GetName()
+
+			if funk.Contains(usedConfigMaps, resource) {
+				// already marked as existing, skip this
+				return
+			}
+			contains, err := cms.helper.ResourceContains(namespace, resourceName, predefinedResource)
+			if err != nil {
+				funcErr = err
+				return
+			}
+
+			if contains {
+				usedConfigMaps = append(usedConfigMaps, resource)
+			}
+		})
+	})
+
+	for _, resource := range configMaps {
+		if !funk.Contains(usedConfigMaps, resource) {
+			unusedConfigMaps = append(unusedConfigMaps, resource)
+		}
+	}
+
+	return unusedConfigMaps, funcErr
+}
+
+// Delete removes Config Maps
+func (cms ConfigMapsService) Delete(configMaps []v1.ConfigMap) {
+	for _, resource := range configMaps {
+		namespace := resource.GetNamespace()
+		name := resource.GetName()
+
+		log.Infof("Deleting config maps %s/%s", namespace, name)
+
+		err := cms.client.Delete(name, &metav1.DeleteOptions{})
+
+		if err != nil {
+			log.WithError(err).Errorf("Failed to delete config map %s/%s", namespace, name)
+		}
+	}
+}
+
+//FilterByTime returns config maps which are older than specified date
+func (cms ConfigMapsService) FilterByTime(configMaps []v1.ConfigMap, olderThan time.Time) (filteredResources []v1.ConfigMap) {
+	log.WithFields(log.Fields{
+		"olderThan":  olderThan,
+		"configMaps": configMaps,
+	}).Debug("Filtering resources older than the specified time")
+
+	for _, resource := range configMaps {
+		lastUpdatedDate := resource.GetCreationTimestamp().Time
+		if lastUpdatedDate.Before(olderThan) {
+			filteredResources = append(filteredResources, resource)
+		} else {
+			name := resource.GetName()
+			log.WithField("name", name).Debugf("Filtered resource %s", name)
```

Comment:
The name is doubled in the log. Please decide whether you include the name in the logging field, or format in the log message itself. So either do `log.WithField("name", resource.GetName()).Debug("Filtered resource")` or `log.Debugf("Filtered resource %s", resource.GetName())`